File: java/ems_java/src/br/erlangms/IEmsRequest.java
/*********************************************************************
 * @title Interface IEmsRequest
 * @version 1.0.0
 * @doc Interface que representa uma requisição para um serviço
 * @author Everton de Vargas Agilar <evertonagilar@gmail.com>
 * @copyright ErlangMS Team
 ********************/
  
package br.erlangms;

import java.text.ParseException;
import java.util.Date;
import java.util.Map;

import com.ericsson.otp.erlang.OtpErlangObject;

public interface IEmsRequest {
	public long getRID();
	public String getUrl();
	public String getMetodo();
	public int getParamsCount();
	public String getParam(final String NomeParam);
	public int getParamAsInt(final String NomeParam);
	public double getParamAsDouble(final String NomeParam);
	public int getQueryCount();
	public String getQuery(final String Nome);
	public int getQueryAsInt(final String Nome);
	public double getQueryAsDouble(final String Nome);
	public String getModulo();
	public String getFunction();
	public OtpErlangObject getOtpRequest();
	public Date getParamAsDate(final String NomeParam) throws ParseException;
	public String getPayload();
	public <T> T getObject(Class<T> classOfObj, EmsJsonModelAdapter jsonModelAdapter);	
	public <T> T getObject(Class<T> classOfObj);
	public Map<String, Object> getObject();
	public Object getProperty(final String nome);
	public Integer getPropertyAsInt(final String nome);
	public Object mergeObjectFromPayload(Object obj);
	public Object mergeObjectFromPayload(Object obj, EmsJsonModelAdapter jsonModelAdapter);
	public Map<String, Object> getPayloadAsMap();
	public String getContentType();
}

File: berco/Cliente/Cliente/IEmsRequest.cs
using System;
using Erlang.NET;
using System.Collections.Generic;

namespace br.erlangms
{
	public interface IEmsRequest<T>
	{
		long getRID();
		string getUrl();
		string getMetodo();
		int getParamsCount();
		string getParam(String NomeParam);
		int getParamAsInt(String NomeParam);
		double getParamAsDouble(String NomeParam);
		int getQueryCount();
		string getQuery(String Nome);
		int getQueryAsInt(String Nome);
		double getQueryAsDouble(String Nome);
		string getModulo();
		string getFunction();
		OtpErlangObject getOtpRequest();
		DateTime getParamAsDate(String NomeParam);
		String getPayload();
		T getObject(Type classOfObj, EmsJsonModelAdapter jsonModelAdapter);
		T getObject(Type classOfObj);
		Dictionary<String,Object> getObject();
		Object getProperty(String nome);
		int getPropertyAsInt(String nome);
		Object mergeObjectFromPayload(Object obj);
		Object mergeObjectFromPayload(Object obj, EmsJsonModelAdapter jsonModelAdapter);
		Dictionary<String, Object> getPayloadAsMap();
		String getContentType();
	}
}

